velox/expression/CastExpr.cpp

```diff
@@ -838,6 +858,41 @@ VectorPtr CastExpr::applyTimestampToVarcharCast(
   return result;
 }
 
+template <typename TInput>
+VectorPtr CastExpr::applyIntToBinaryCast(
+    const SelectivityVector& rows,
+    exec::EvalCtx& context,
+    const BaseVector& input) {
+  VELOX_USER_CHECK(
+      hooks_->canCastIntToBinary(),
+      "Cannot cast {} to VARBINARY.",
```

Comment:
It seems too late for this type of check. Shouldn't we fail much earlier when compiling the expression and creating an instance of CastExpr?

CC: @kagamiori

Reply:
Moved this check to constructSpecialForm which will be called in compileRewrittenExpression.
https://github.com/facebookincubator/velox/blob/b842c14fe10d17cedcbb9c59b196c7be56122b4b/velox/expression/ExprCompiler.cpp#L408-L409